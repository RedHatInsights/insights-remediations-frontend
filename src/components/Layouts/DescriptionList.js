import React from 'react';
import propTypes from 'prop-types';
import classnames from 'classnames';

import './DescriptionList.scss';

const DescriptionList = ({ title, className, children, isBold, ...props }) => {

    const descriptionListClasses = classnames(
        'ins-l-description-list__description',
        { ['ins-l-description-list__description--bold']: isBold }
    );

    const mainClasses = classnames(
        className,
        'ins-l-description-list'
    );

    return (
        <dl className={ mainClasses } { ...props }>
            <dt className='ins-l-description-list__title'><b>{ title }</b></dt>
            <dd className={ descriptionListClasses }> { children } </dd>
        </dl>
    );
};

export default DescriptionList;

DescriptionList.propTypes = {
    title: propTypes.string,
    children: propTypes.any,
    isBold: propTypes.bool,
    className: propTypes.string
};
