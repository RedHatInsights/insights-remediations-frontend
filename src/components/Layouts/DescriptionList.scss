.ins-l-description-list {
    .ins-l-description-list__title {
        margin-bottom: 6px; // form-element-spacer
        font-size: var(--pf-global--FontSize--sm);
    }
    .ins-l-description-list__description {
        font-size: var(--pf-global--FontSize--md);
        &--bold {
            font-size: var(--pf-global--FontSize--lg);
            font-weight: var(--pf-global--FontWeight--redhatfont--bold);
        }
    }
    .ins-l-description-list__description--withGutter {
        * + * { margin-left: var(--pf-global--spacer--sm); }
    }
    .ins-l-description-list__description--needsPointer {
        cursor: pointer;
    }
}