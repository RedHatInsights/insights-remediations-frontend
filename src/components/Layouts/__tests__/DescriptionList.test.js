import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DescriptionList from '../DescriptionList';

describe('DescriptionList component', () => {

    it('should render', () => {
        const wrapper = shallow(<DescriptionList title='title'>DescriptionList</DescriptionList>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with bold', () => {
        const wrapper = shallow(<DescriptionList isBold title='title'>DescriptionList</DescriptionList>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
