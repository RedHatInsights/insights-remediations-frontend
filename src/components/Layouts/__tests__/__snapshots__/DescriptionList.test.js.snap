// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DescriptionList component should render 1`] = `
<dl
  className="ins-l-description-list"
>
  <dt
    className="ins-l-description-list__title"
  >
    <b>
      title
    </b>
  </dt>
  <dd
    className="ins-l-description-list__description"
  >
     
    DescriptionList
     
  </dd>
</dl>
`;

exports[`DescriptionList component should render with bold 1`] = `
<dl
  className="ins-l-description-list"
>
  <dt
    className="ins-l-description-list__title"
  >
    <b>
      title
    </b>
  </dt>
  <dd
    className="ins-l-description-list__description ins-l-description-list__description--bold"
  >
     
    DescriptionList
     
  </dd>
</dl>
`;
